numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -91,50 +121,40 @@ NPY_NO_EXPORT int
 PyArray_IntpConverter(PyObject *obj, PyArray_Dims *seq)
 {
     Py_ssize_t len;
-    int nd;
+    PyObject *const *items;
+    PyObject *fast_seq = NULL;
 
     seq->ptr = NULL;
     seq->len = 0;
     if (obj == Py_None) {
         return NPY_SUCCEED;
     }
-    len = PySequence_Size(obj);
-    if (len == -1) {
-        /* Check to see if it is an integer number */
-        if (PyNumber_Check(obj)) {
-            /*
-             * After the deprecation the PyNumber_Check could be replaced
-             * by PyIndex_Check.
-             * FIXME 1.9 ?
```

Comment:
This is already handled within `PyArray_IntpFromIndexSequence` on line 131 - all this did was forbid objects that implement `__index__` but neither `__int__` nor `__float__`, and try to produce better error messages.

The deprecation this refers to was eliminated a long time ago.